### PR TITLE
Add Hiera defaults support for module documentation

### PR DESCRIPTION
**Pull request by slauger**
## Summary

This PR adds support for reading parameter defaults from Hiera data files (`data/common.yaml`) when generating module documentation with openvox-strings.

Previously, openvox-strings could only document parameter defaults that were explicitly defined in Puppet code. With this feature, defaults can now be maintained in Hiera data files while still appearing correctly in the generated REFERENCE.md documentation.

## Motivation

Many Puppet modules use Hiera for default values to enable easier customization and separation of data from code. However, this created a documentation gap - parameters without code defaults wouldn't show any default value in the generated documentation, even though they had defaults in `data/common.yaml`.

### Core Changes

1. **New `OpenvoxStrings::Hiera` class** (`lib/openvox-strings/hiera.rb`)
   - Parses `hiera.yaml` configuration
   - Loads data from `data/common.yaml`
   - Provides `lookup_default(class_name, param_name)` method
   - Converts Ruby values to Puppet-compatible string representation

2. **Integration in `OpenvoxStrings::Markdown::Base`** (`lib/openvox-strings/markdown/base.rb`)
   - Modified `defaults` method to merge code defaults with Hiera defaults
   - Code defaults take precedence (backwards compatible)
   - Automatically detects module root from manifest file paths
   - Only activates when `hiera.yaml` exists

### Key Features

- ✅ **Backwards compatible**: Code defaults always take precedence over Hiera defaults
- ✅ **Zero configuration**: Automatically detects and uses Hiera if `hiera.yaml` exists
- ✅ **All data types supported**: String, Integer, Float, Boolean, Array, Hash, nil/undef, nested structures
- ✅ **Exact formatting match**: Output is byte-for-byte identical to code defaults format

### Data Type Conversions

The implementation correctly handles all Puppet data types:
- Strings: `'value'` (quoted)
- Numbers: `42`, `3.14` (unquoted)
- Booleans: `true`, `false` (lowercase, unquoted)
- nil/undef: `undef`
- Empty collections: `[]`, `{}`
- Arrays: `['a', 'b', 'c']` (no spaces around brackets to match puppet-strings format)
- Hashes: `{ 'key' => 'value' }` (spaces inside braces)
- Nested structures: Recursively converted

### Unit Tests

Comprehensive test suite in `spec/unit/openvox-strings/hiera_spec.rb` covering:
- Hiera configuration loading
- Data file parsing
- All Puppet data types
- Nested structures
- Edge cases (missing files, wrong module names, etc.)
- Array formatting without spaces to match code defaults

### Integration Tests

Three-stage integration test in `slauger/puppet-github_actions_runner`:

1. **[Test 1 - Baseline (PR#4)](https://github.com/slauger/puppet-github_actions_runner/pull/4)**
   - Uses OLD openvox-strings (6.0.0 from RubyGems)
   - All defaults in code
   - Establishes baseline REFERENCE.md

2. **[Test 2 - Backwards Compatibility (PR#5)](https://github.com/slauger/puppet-github_actions_runner/pull/5)**
   - Uses NEW openvox-strings (this PR)
   - All defaults still in code
   - Proves 100% backwards compatibility

3. **[Test 3 - Full Hiera (PR#6)](https://github.com/slauger/puppet-github_actions_runner/pull/6)**
   - Uses NEW openvox-strings (this PR)
   - ALL defaults moved from code to `data/common.yaml`
   - Proves Hiera defaults feature works correctly

**Result**: All three tests produce byte-for-byte identical REFERENCE.md documentation.

### Test Data Types Covered

The integration tests validate all common Puppet data types:
- `Integer`: `42`
- `Float`: `3.14`
- `Array` (empty): `[]`
- `Array[String]`: `['alpha', 'beta', 'gamma']`
- `Array[Integer]`: `[1, 2, 3, 5, 8]`
- `Hash` (nested): `{ 'config' => { 'timeout' => 30, 'retry' => true }, 'tags' => ['prod', 'eu'] }`
- Plus all standard module parameters (Strings, Booleans, Optional types, etc.)

## Breaking Changes

None. This is a fully backwards-compatible addition:
- Modules without Hiera: No change in behavior
- Modules with Hiera but code defaults: Code defaults take precedence
- Only affects modules that use Hiera defaults WITHOUT code defaults

## Documentation

The implementation is self-documenting through:
- YARD documentation in code
- Comprehensive unit tests showing usage
- Integration test examples in real-world module

## Related Issues

This addresses the common pain point where modules using Hiera v5 data-in-modules pattern couldn't document their defaults properly.

### CODE REVIEW

**Review by ekohl — COMMENTED** [review]
https://github.com/puppetlabs/puppet-strings/pull/273 was my attempt a few years ago but I never found the time to finish it. It took a bigger approach of also expanding facts so you got a complete table.

**Comment by slauger** [comment]
@ekohl any feedback on the latest changes?

File: lib/openvox-strings/hiera.rb [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    def load_hiera_config
+      hiera_file = File.join(@module_path, 'hiera.yaml')
+      return nil unless File.exist?(hiera_file)
+
+      begin
+        YAML.load_file(hiera_file)
+      rescue StandardError => e
+        YARD::Logger.instance.warn "Failed to parse hiera.yaml: #{e.message}"
+        nil
+      end
```
Comment (ekohl) [inline lib/openvox-strings/hiera.rb]: This can be simplified by reusing `load_yaml_data`:
```suggestion
      load_yaml_data(hiera_file)
```

**Review by ekohl — APPROVED** [review]
From reading the code I think this looks correct. There are 2 minor things inline, but that shouldn't hold back the merge.

Anyone else would like to weigh in before we merge this?

File: lib/openvox-strings/hiera.rb [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+      # Find first hierarchy entry without interpolations
+      hierarchy = @hiera_config['hierarchy']
+      return nil unless hierarchy
+
+      first_static = hierarchy.find do |entry|
+        path = entry['path'] || entry['paths']&.first
+        next false unless path
+
+        # Check if path contains interpolations like %{...}
+        !path.include?('%{')
```
[the remark this answers is not in the document]
Comment (ekohl) [reply lib/openvox-strings/hiera.rb]: Thinking further, `paths` can be an array and then `.include?` won't crash but `Array#include?` wouldn't work the way you expect. So probably something like:
```suggestion
        case path
        when String
          !path.include?('%{')
        when Array
          path.none? { |p| p.include?('%{') }
        else
          false
        end
```

[the remark this answers is not in the document]
Comment (slauger) [reply lib/openvox-strings/hiera.rb]: Now I slowly understand why you never finished your PR 😄

Can you provide a test that proves the current implementation is broken? I'm not seeing the problem.

**Comment by bastelfreak** [comment]
Hi @slauger . I gave this a test in https://github.com/voxpupuli/puppet-systemd/pull/607. I regenerated the REFERENCE.md with `bundle exec rake strings:generate:reference` but that didn't change the REFERENCE.md. is there something else that needs to be configured? Did you test this in a module?

**Comment by slauger** [comment]
@bastelfreak the first match in the `hiera.yaml` of `puppet-systemd` w/o interpolation is a `common.yaml`.

https://github.com/voxpupuli/puppet-systemd/blob/master/hiera.yaml#L15

But there is no `common.yaml` inside the data directory, so no changes are expected. For PRs with test data checkout the links in the first post.

**Comment by kenyon** [comment]
Just mentioning that this implemented https://github.com/puppetlabs/puppet-strings/issues/250 so that there's a link back here.
